src/parquet/thrift/util.h

```diff
@@ -34,13 +35,44 @@ inline void DeserializeThriftMsg(const uint8_t* buf, uint32_t* len, T* deseriali
       tproto_factory.getProtocol(tmem_transport);
   try {
     deserialized_msg->read(tproto.get());
-  } catch (apache::thrift::protocol::TProtocolException& e) {
-    throw ParquetException("Couldn't deserialize thrift.", e);
+  } catch (std::exception& e) {
+    //TODO Determine an appproach to log exceptions
+    LOG(1) << "Couldn't deserialize thrift: " << e.what();
+    return false;
```

Comment:
Hm, so this is halfway in between Impala (which uses impala::Status to communicate success/failure) and parquet_cpp (which uses exceptions). Why not raise the exception and catch it in `serialized-page.cc`, rethrowing when we exceed the maximum allowed page size?